Tests/DatadogTests/Datadog/RUM/RUMVitals/VitalRefreshRateReaderTests.swift

```diff
@@ -0,0 +1,78 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import XCTest
+import UIKit
+@testable import Datadog
+
+class VitalRefreshRateReaderTests: XCTestCase {
+    private let mockNotificationCenter = NotificationCenter()
+
+    func testHighAndLowRefreshRates() {
+        let reader = VitalRefreshRateReader(notificationCenter: mockNotificationCenter)
+        let observer_view1 = VitalObserver(listener: VitalListenerMock())
+        let observer_view2 = VitalObserver(listener: VitalListenerMock())
+
+        // View1 has simple UI, high FPS expected
+        reader.register(observer_view1)
+
+        // Wait without blocking UI thread
+        let expectation1 = expectation(description: "async expectation for first observer")
+        DispatchQueue.global().asyncAfter(deadline: .now() + 0.5) {
+            expectation1.fulfill()
+        }
+
+        waitForExpectations(timeout: 1.0) { _ in
+            XCTAssertGreaterThan(observer_view1.vitalInfo.sampleCount, 0)
+            XCTAssertGreaterThan(UIScreen.main.maximumFramesPerSecond, Int(observer_view1.vitalInfo.maxValue))
+            XCTAssertGreaterThan(observer_view1.vitalInfo.minValue, 0.0)
+        }
+
+        reader.unregister(observer_view1)
+        // View2 has complex UI, lower FPS expected
+        reader.register(observer_view2)
+
+        // Wait without blocking UI thread
+        let expectation2 = expectation(description: "async expectation for second observer")
+        DispatchQueue.global().asyncAfter(deadline: .now() + 0.5) {
+            expectation2.fulfill()
+        }
+        waitForExpectations(timeout: 1.0) { _ in }
+
+        // Block UI thread
+        Thread.sleep(forTimeInterval: 1.5)
+
+        XCTAssertGreaterThan(observer_view2.vitalInfo.sampleCount, 0)
+        XCTAssertGreaterThan(observer_view1.vitalInfo.meanValue, observer_view2.vitalInfo.meanValue)
```

Comment:
this line is flaky. i'm looking into it...

Reply:
fixed: now we first block UI thread (lowering FPS) and then wait in background thread so that reader can read FPS **BEFORE** `XCTAssert`s